lib/_http_outgoing.js

```diff
@@ -728,8 +729,28 @@ OutgoingMessage.prototype.setHeaders = function setHeaders(headers) {
     throw new ERR_INVALID_ARG_TYPE('headers', ['Headers', 'Map'], headers);
   }
 
-  for (const key of headers.keys()) {
-    this.setHeader(key, headers.get(key));
+  // Headers object joins multiple cookies with a comma when using
+  // the getter to retrieve the value,
+  // unless iterating over the headers directly.
+  // We also cannot safely split by comma.
+  // To avoid setHeader owerwriting the previous value we push
+  // set-cookie values in array and set them all at once.
+  const headersToSplit = new SafeSet(['set-cookie']);
```

Comment:
I'm a bit lost on why this is a set. It seems to have only one value in it

Reply:
Yeah, maybe its not needed and can be simplified, just wondering if there are any other case other than `set-cookie`

Reply:
I would recommend simplification.